### PR TITLE
Tests/registration

### PR DESCRIPTION
**Pull request by dasyad00**
add tests for registration views

## Motivation and Context
missing tests

### CODE REVIEW

File: apps/users/models/user/profile.py [hunk 1]
```diff
@@ -36,7 +36,8 @@ class Profile(models.Model):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.states = ["subscribed_newsletter", "newsletter_recipient"]
-        self.update_states()
+        # TODO fix states
+        # self.update_states()
```
Comment (bcvandendool) [inline apps/users/models/user/profile.py:40]: I had some unfinished work on unit tests, which included starting to fix this. I pushed it to the feature/unit-test branch so you can take a look at it, but not sure what the current state is. It also contains some fixed up unit tests for the async_requests and some for users. If you prefer, I can also take a look at it later, but that might still be a while.

File: tests/user/forms/registration.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    def test_success_tue(self):
+        """
+        TUe User registration success
+        """
+        # setup
+        exp_institution = "tue"
+
+        # act
+        form = self.generate_form()
+        user: User = form.save()
```
Comment (bcvandendool) [inline tests/user/forms/registration.py:47]: Either add types or not, but at least be consistent in one file.

File: tests/user/views/registration.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                    "username": "tosti@fontys.nl",
+                    "email": "tosti@gmail.com",
+                    "password1": "ikbeneenbrood",
+                    "password2": "ikbeneenbrood",
+                    "nationality": "Dutch",
+                    "terms_confirmed": "on",
+                    "subscribed_newsletter": "on",
+                },
+                "register_fontys": {
+                    "study": "test",  # TODO fix Fontys studies
```
Comment (bcvandendool) [inline tests/user/views/registration.py:94]: Please add an issue on github for this so we dont forget

File: tests/user/views/registration.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
+from bs4 import BeautifulSoup
+from django.contrib.auth.models import User
+from django.core import mail
+
+from tests.user.views.wizard_helper import WizardViewTestCase
+
+
```
Comment (bcvandendool) [inline tests/user/views/registration.py:7]: Maybe add tests to make sure terms and conditions are agreed to, and that the newsletter preferences is set correctly, and also correctly updated on sendgrid, as those two things are quite important?

**Review by bcvandendool — CHANGES_REQUESTED** [review]

File: tests/user/views/registration.py (hunk 4 above)
Comment (dasyad00) [reply tests/user/views/registration.py:7]: from what I can tell, the check is done client side via javascript but I'm not sure how to confirm this. mind looking into it?

File: apps/users/models/user/profile.py (hunk 1 above)
Comment (dasyad00) [reply apps/users/models/user/profile.py:40]: I plan to create tests for the update views and models in a separate PR

Comment (bcvandendool) [reply apps/users/models/user/profile.py:40]: fair enough

File: tests/user/views/registration.py (hunk 4 above)
Comment (bcvandendool) [reply tests/user/views/registration.py:7]: afaik no js is involved, all validation is done in python, and it renders back the view if it is invalid no? Or do you mean the required tag? Cause that is just plain html, but a test should be made anyway, cause we want to validate correct behaviour for this.

**Review by bcvandendool — APPROVED** [review]

**Pull request merged**